packages/hardhat-core/src/internal/hardhat-network/provider/provider.ts

```diff
@@ -438,6 +445,10 @@ export class EdrProviderWrapper
       };
     }
 
+    // To accomodate construction ordering, we need an adapter to listen to
+    // events from the provider's subscriber callback
```

Comment:
I'm not sure I understand this comment.

Reply:
Ok, I think I get it now. Maybe I would say something like "we need an adapter to forward events from the EdrProvider callback to the wrapper's listener". I'm not sure if that would be an improvement though. Anyway, feel free to ignore this.

Reply:
Incorporated in https://github.com/NomicFoundation/hardhat/pull/4747/commits/10606d8809530f7b8f6023d595ffef07d13197c9